$NetBSD: patch-ac,v 1.25 2016/12/30 11:28:19 wiz Exp $

Allow ncurses to build with compiler from pkgsrc by dropping the use of
-no-cpp-precomp which is specific to Apple's supplied toolchain.
Don't rename man pages on Debian.
Include local paths before utilising flags from build env

--- configure.orig	2015-08-05 09:20:32.000000000 +0000
+++ configure
@@ -4930,7 +4930,7 @@ test -z "$cf_cv_libtool_version" && unse
 echo "$as_me: error: Cannot find libtool" >&2;}
    { (exit 1); exit 1; }; }
 	fi
-	LIB_CREATE='${LIBTOOL} --mode=link ${CC} -rpath ${DESTDIR}${libdir} ${LIBTOOL_VERSION} `cut -f1 ${top_srcdir}/VERSION` ${LIBTOOL_OPTS} ${LT_UNDEF} $(LIBS) -o'
+	LIB_CREATE='${LIBTOOL} --mode=link ${CC} -rpath ${libdir} ${LIBTOOL_VERSION} `cut -f1 ${top_srcdir}/VERSION` ${LIBTOOL_OPTS} ${LT_UNDEF} $(LIBS) -o'
 	LIB_OBJECT='${OBJECTS:.o=.lo}'
 	LIB_SUFFIX=.la
 	LIB_CLEAN='${LIBTOOL} --mode=clean'
@@ -11090,12 +11090,7 @@ case ".$MANPAGE_RENAMES" in
 (.no)
 	;;
 (.|.yes)
-	# Debian 'man' program?
-	if test -f /etc/debian_version ; then
-		MANPAGE_RENAMES=`cd $srcdir && pwd`/man/man_db.renames
-	else
-		MANPAGE_RENAMES=no
-	fi
+	MANPAGE_RENAMES=no
 	;;
 esac
 
@@ -21894,8 +21889,16 @@ EOF
 #define HAVE_LIB${cf_have_include} 1
 EOF
 
-			TEST_DEPS="${LIB_DIR}/${LIB_PREFIX}${cf_dir}${DFT_DEP_SUFFIX} $TEST_DEPS"
-			TEST_DEP2="${LIB_2ND}/${LIB_PREFIX}${cf_dir}${DFT_DEP_SUFFIX} $TEST_DEP2"
+			case ${cf_dir} in
+			form|menu|panel)
+				TEST_DEPS="${LIB_DIR}/${LIB_PREFIX}gnu${cf_dir}${DFT_DEP_SUFFIX} $TEST_DEPS"
+				TEST_DEP2="${LIB_2ND}/${LIB_PREFIX}gnu${cf_dir}${DFT_DEP_SUFFIX} $TEST_DEP2"
+				;;
+			*)
+				TEST_DEPS="${LIB_DIR}/${LIB_PREFIX}${cf_dir}${DFT_DEP_SUFFIX} $TEST_DEPS"
+				TEST_DEP2="${LIB_2ND}/${LIB_PREFIX}${cf_dir}${DFT_DEP_SUFFIX} $TEST_DEP2"
+				;;
+			esac
 			if test "$DFT_LWR_MODEL" = "libtool"; then
 				TEST_ARGS="${TEST_DEPS}"
 				TEST_ARG2="${TEST_DEP2}"
@@ -22181,9 +22184,9 @@ fi
 ADAHTML_DIR=../../doc/html/ada
 
 # these could be configurable, but are not currently
-PANEL_NAME=panel
-MENU_NAME=menu
-FORM_NAME=form
+PANEL_NAME=gnupanel
+MENU_NAME=gnumenu
+FORM_NAME=gnuform
 CXX_NAME=ncurses++
 
 # workaround for g++ versus Solaris (20131116)
@@ -23565,7 +23568,14 @@ CF_EOF
 		cf_depsuf="${LIB_SUFFIX}${EXTRA_SUFFIX}${cf_depsuf}"
 	fi
 
-			cf_libname=$cf_dir
+			case $cf_dir in
+			form|menu|panel)
+				cf_libname=gnu$cf_dir
+				;;
+			*)
+				cf_libname=$cf_dir
+				;;
+			esac
 			test "$cf_dir" = c++ && cf_libname=ncurses++
 			if test $cf_item = shared ; then
 				if test -n "${LIB_SUFFIX}"
